core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java

```diff
@@ -559,6 +580,11 @@ public Map<String,String> getHints() {
               "For tablet {} scan server selector chose scan_server:{} delay:{} busyTimeout:{}",
               loc.tablet_extent, scanServer, delay, scanState.busyTimeout);
         } else {
+          try {
+            context.requireNotOffline(scanState.tableId, context.getTableName(scanState.tableId));
+          } catch (TableNotFoundException e) {
+            throw new RuntimeException("Error scanning for table: " + scanState.tableId, e);
```

Comment:
Some of these generic RTEs might be better expressed as IllegalStateException. In general, it's preferable to use a more specific RTE whenever possible.

Reply:
Created UncheckedAccumuloException in 5b98e2a